### PR TITLE
🐛 Salesforce Connector: handling 400 error, while sobject doesn't support query or queryAll requests

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Added error handling when reading streams that do not support requests for `query` or `queryAll` from Salesforce API.
Resolves https://github.com/airbytehq/oncall/issues/79*

## How
*Catching errors using the `errorCode` parameter in the response from Salesforce API*

#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### CODE REVIEW

**Comment by yevhenii-ldv** [comment]
I can't run the tests, because the limit on requests for today has run out.

**Review by htrueman — APPROVED** [review]

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 1]
```diff
@@ -90,9 +90,21 @@ def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
...
+            """
             error_data = error.response.json()[0]
-            if error.response.status_code == codes.FORBIDDEN and not error_data.get("errorCode", "") == "REQUEST_LIMIT_EXCEEDED":
-                self.logger.error(f"Cannot receive data for stream '{self.name}', error message: '{error_data.get('message')}'")
+            if error.response.status_code in [codes.FORBIDDEN, codes.BAD_REQUEST]:
+                error_code = error_data.get("errorCode", "")
+                if error_code != "REQUEST_LIMIT_EXCEEDED" or error_code == "INVALID_TYPE_FOR_OPERATION":
+                    self.logger.error(f"Cannot receive data for stream '{self.name}', error message: '{error_data.get('message')}'")
+                else:
+                    raise error
```
Comment (Zirochkaa) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:107]: You can remove two occurrences of `else: raise error` code and the code will look like this:
```
     error_data = error.response.json()[0]
     if error.response.status_code in [codes.FORBIDDEN, codes.BAD_REQUEST]:
         error_code = error_data.get("errorCode", "")
         if error_code != "REQUEST_LIMIT_EXCEEDED" or error_code == "INVALID_TYPE_FOR_OPERATION":
             self.logger.error(f"Cannot receive data for stream '{self.name}', error message: '{error_data.get('message')}'")
             return
     raise error
```

**Review by Zirochkaa — APPROVED** [review]

**Review by sergei-solonitcyn — COMMENTED** [review]
Let's apply changes proposed by Oleh after double-check. And I think, we could merge it.

**Pull request merged**